api/client/gro_client.py

```diff
@@ -126,9 +131,25 @@ def add_data_series(self, **kwargs):
             data_series_list = self.get_data_series(**entities)
             self._logger.debug("Found {} distinct data series for {}".format(
                 len(data_series_list), entities))
+            # temporal coverage affects ranking so add time range if specified.
+            for data_series in data_series_list:
+                if kwargs.get('start_date'):
+                    data_series['start_date'] = kwargs['start_date']
+                if kwargs.get('end_date'):
+                    data_series['end_date'] = kwargs['end_date']
             for data_series in self.rank_series_by_source(data_series_list):
```

Comment:
Do note rank_series_by_source() still doesn't work as expected when passing in a full data series. https://grointelligence.atlassian.net/browse/CLEWS-17127

Only accepts metric/item/region/frequency. No source_id.

Reply:
Right. Actuall it does accept (and correctly ignore) source_id but  the source_name remains in the output and mismatches. Fixed in lib.py